src/models/indexer_metadata.rs

```diff
@@ -0,0 +1,177 @@
+use serde::Deserialize;
+use sqlx::{Pool, Postgres};
+use std::sync::Arc;
+use tracing::error;
+
+use super::model::ModelError;
+
+#[derive(Debug, Deserialize, sqlx::FromRow)]
+pub struct IndexMetadata {
+    pub id: i64,
+    pub current_latest_block_number: i64,
+    pub indexing_starting_block_number: i64,
+    pub is_backfilling: bool,
+    pub updated_at: chrono::DateTime<chrono::Utc>,
+}
+
+pub trait IndexMetadataModelTrait {
+    async fn get_index_metadata(&self) -> Result<IndexMetadata, ModelError>;
+    async fn set_is_backfilling(&self, is_backfilling: bool) -> Result<(), ModelError>;
+    async fn set_initial_indexing_status(
+        &self,
+        current_latest_block_number: i64,
+        indexing_starting_block_number: i64,
+        is_backfilling: bool,
+    ) -> Result<(), ModelError>;
+    async fn update_latest_quick_index_block_number(
+        &self,
+        block_number: i64,
+    ) -> Result<(), ModelError>;
+}
+
+// Model is used to interact with the database
+pub struct IndexMetadataModel(Arc<Pool<Postgres>>);
+
+impl IndexMetadataModelTrait for IndexMetadataModel {
+    async fn get_index_metadata(&self) -> Result<IndexMetadata, ModelError> {
+        let result: IndexMetadata = sqlx::query_as(
+            r#"
+            SELECT id, current_latest_block_number, indexing_starting_block_number, is_backfilling, updated_at
+            FROM indexer_metadata
+            "#,
+        )
+        .fetch_one(&*self.0)
+        .await?;
+
+        Ok(result)
+    }
+
+    async fn set_is_backfilling(&self, is_backfilling: bool) -> Result<(), ModelError> {
+        let result = sqlx::query(
+            r#"
+            UPDATE indexer_metadata
```

Comment:
In all this UPDATE are you not using WHERE intentionally? Without it seems that you want to update all the rows on the table?

Reply:
in this case, yes. Since there can only (and should only) be one row for this table, this should be acceptable. 

If you foresee some cases where this might be held false, do tell me about it.